### ISSUE TITLE
adapter-cloudflare does not bundle service-worker.js since around next.207 - results in 404 and thus failed SW registration

### ISSUE THREAD

**Issue by codepainting** [sveltejs/kit#3187]
### Describe the bug

Since around next.207 `adapter-cloudflare` seems not to bundle `service-worker.js`

When testing locally, everything seems OK. The SW file is present in the `./svelte-kit/cloudflare` builld folder - as it was with older versions of Sveltekit. When running `npm run preview` the SW is being successfully registered according to Chrome Dev Tools.

After deployment﻿ to CF, and running the site, the following error shows in the console log:
`Uncaught (in promise) TypeError: Failed to register a ServiceWorker for scope ('https://sw-cloudflare-repro.pages.dev/') with script ('https://sw-cloudflare-repro.pages.dev/service-worker.js'): A bad HTTP response code (404) was received when fetching the script.`

Also, according to Dev Tools, the SW is not present.

### Reproduction

[https://github.com/codepainting/sw-cloudflare-repro](https://github.com/codepainting/sw-cloudflare-repro)

Basically it's the standard demo app, with the following additions:
+ `service-worker.js` is included in `/src`
+ `import adapter from '@sveltejs/adapter-cloudflare';` is configured

Demo site:
[https://sw-cloudflare-repro.pages.dev/](https://sw-cloudflare-repro.pages.dev/)

Run demo site.
Look into the console logs.

### Logs

```shell
Uncaught (in promise) TypeError: Failed to register a ServiceWorker for scope ('https://sw-cloudflare-repro.pages.dev/') with script ('https://sw-cloudflare-repro.pages.dev/service-worker.js'): A bad HTTP response code (404) was received when fetching the script.
```

### System Info

```shell
System:
    OS: Linux 4.19 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (4) x64 Intel(R) Core(TM) i5-7600K CPU @ 3.80GHz
    Memory: 19.42 GB / 25.03 GB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.volta/tools/image/node/16.13.0/bin/node
    Yarn: 1.22.4 - ~/.volta/bin/yarn
    npm: 8.1.4 - ~/.volta/tools/image/npm/8.1.4/bin/npm
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.7 
    @sveltejs/adapter-cloudflare: ^1.0.0-next.5 => 1.0.0-next.5 
    @sveltejs/kit: next => 1.0.0-next.213 
    svelte: ^3.44.0 => 3.44.3
```

### Severity

blocking an upgrade

**Comment by CobyPear** [comment]
This behavior is happening in adapter-vercel as well.

**Comment by Mooshua** [comment]
The issue seems to be with the internal manifest of the workers.

```js
export const manifest = {
	appDir: "_app",
	assets: new Set(["favicon.png", /*removed for brevity...*/,"svg/Youtube.svg","svg/YoutubeRed.svg" /* no service-worker.js :( */]),
	_: {
		mime: {".png":"image/png",".jpg":"image/jpeg",".svg":"image/svg+xml"},
		entry: {"file":"start-d1b74b63.js","js":["start-d1b74b63.js","chunks/vendor-873c0a70.js"],"css":["assets/start-d5b4de3e.css"]},
		nodes: [
			() => import('../output/server/nodes/0.js'),
			() => import('../output/server/nodes/1.js'),
			() => import('../output/server/nodes/2.js'),
			() => import('../output/server/nodes/3.js'),
			() => import('../output/server/nodes/4.js'),
			() => import('../output/server/nodes/5.js'),
			() => import('../output/server/nodes/6.js'),
			() => import('../output/server/nodes/7.js'),
			() => import('../output/server/nodes/8.js'),
			() => import('../output/server/nodes/9.js'),
			() => import('../output/server/nodes/10.js')
		],
		routes: [
			{
				type: 'page',
				pattern: /^\/$/,
				params: null,
				path: "/",
				a: [0,2],
				b: [1]
			},
			/*...*/
			]
```
This is created from `builder.generateManifest` ([link to impl](https://github.com/sveltejs/kit/blob/da29c3fb01b43674c9a824efcd983047ef8e488c/packages/kit/src/core/generate_manifest/index.js#L10)), and does *not* seem to include service workers.

**Comment by Envl** [comment]
I stumbled into the same issue  a few days ago. My current workaround is to add an additional `service-worker.js` into  folder `/static`, such that the generated service worker will be included in the final bundle.   

(Yes, **THE SERVICE WORKER** being served is compiled from `src/service-worker.ts` rather than `static/service-worker.js`)  

the demo: https://pwa-test-1mm.pages.dev/service-worker.js  
the repo: https://github.com/Envl/pwa-test    
Notice the code in `static/service-worker.js` is different from what that is in `src/service-worker.ts`. It seems what's inside  `static/service-worker.js` doesn't matter as long as it exists.

**Comment by codepainting** [comment]
Great to see work dedicated to this issue. Unfortunately, the update seems to have no effect - at least in the case of Cloudflare, which I was testing thoroughly (different browsers to avoid SW caching problems).

**Reproduction:**
I updated my initial demo site with a clean new version of the Sveltekit demo app.

When the service-worker.js file is included in /src _alone,_ the following error persists:

```
A bad HTTP response code (404) was received when fetching the script.
sw-cloudflare-repro.pages.dev/:1 Uncaught (in promise) TypeError: Failed to register a ServiceWorker for scope ('https://sw-cloudflare-repro.pages.dev/') with script ('https://sw-cloudflare-repro.pages.dev/service-worker.js'): A bad HTTP response code (404) was received when fetching the script.

```
**But** I can confirm that the workaround proposed by @Envl is working also with the updated Sveltekit.

I added another copy of the service-worker.js file into the /static folder and console logged respectively to see which version is being used when deployed on Cloudflare.

I can confirm that the sheer existence of a service-worker.js file in the static folder somehow triggers the use of the SW, and it is     the one from /src and _not_ from /static.

So while the workaround seems valid also with the new version of Sveltekit, the issue unfortunately seems (at least on Cloudflare) not yet resolved.
